### PR TITLE
doc: DEVELOPMENT.md

### LINKED ISSUE #427: Contributor's guide link in README is a 404
I would have just fixed this, but I can't actually find a contributor's guide or DEVELOPMENT.md to update it to :). We should add one.

### PR DESCRIPTION
**Pull request by lkingland**
Creates a few docs updates, primarily the addition of `DEVELOPMENT.md` which includes the basics for developers to build and test.

Fixes #427

### CODE REVIEW

File: docs/DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+To run core unit tests, use `make test`.
+
+## Linting
+
+Before submitting code in a Pull Request, please run `make check` and resolve any errors.  This creates and runs `bin/golangci-lint`.  For settings such as configured linters, see [.golangci.yaml](../.golangci.yaml).
+
+
+## Templates
+
+When a new Function is created, a few files are placed in the new Function's directory.  This includes source code illustrating a minimal Function of the requested type (language runtime and function signature) as well as Function metadata (`func.yaml`).
```
Comment (lance) [inline docs/DEVELOPMENT.md:24]:
```suggestion
When a new Function is created, a few files are placed in the new Function's directory.  This includes source code illustrating a minimal Function of the requested type (language runtime and function signature) as well as Function metadata in a `func.yaml` file.
```
Nit

File: docs/DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+When changes are made to a template's source code, regenerate `pkged.go` by running `make pkged.go`.  It is also important to run the unit tests of the template modified.  For example, to run the unit tests of the Go templates, use `make test-go`.  For a list of available make targets, use `make help`.
+
+
+## Integration Testing
+
+*Integration tests are run automatically* for all pull requests from within the GitHub Pull Request.  This process includes creating and configuring a lightweight cluster.
+
+If you would like to run integration tests prior to opening a pull request against origin, you can enable Actions in your fork of this repository and create a pull request to your own main branch.
+
+If you would like to run integraiton tests locally, or would like to use the CLI / Client Library directly against a local cluster, the cluster allocation script can be used locally as well:
```
Comment (lance) [inline docs/DEVELOPMENT.md:37]:
```suggestion
If you would like to run integraiton tests locally, or would like to use the CLI / Client Library directly against a local cluster, the cluster allocation script can be used locally as well, by following the steps below.
```
Nit

**Review by lance — COMMENTED** [review]
LGTM -  a couple of minor wording suggestions.

File: docs/DEVELOPMENT.md (hunk 1 above)
Comment (lkingland) [reply docs/DEVELOPMENT.md:24]: Thanks for the good suggestion.  My version was a bit terse and robotic.

File: docs/DEVELOPMENT.md (hunk 2 above)
Comment (lkingland) [reply docs/DEVELOPMENT.md:37]: Again, leaving too much to the imagination; bending semantics to a point that would leave any English teacher squirming.

**Review by lance — APPROVED** [review]
/lgtm

**Comment by julz** [comment]
yay, thanks @lkingland 🎉

**Comment by lkingland** [comment]
> yay, thanks @lkingland 🎉

Thanks for being the impetus @julz!

**Pull request merged**